src/core/translatorstorage.cpp

```diff
@@ -0,0 +1,70 @@
+#include "translatorstorage.h"
+
+#include <QCoreApplication>
+#include <QDebug>
+#include <QDirIterator>
+#include <QLibraryInfo>
+
+TranslatorStorage::TranslatorStorage()
+    : d(std::make_unique<TranslatorStoragePrivate>())
+{}
+
+bool TranslatorStorage::loadAndInstallTranslators(const QString &language)
+{
+    return loadAndInstallTranslators(language, ":/");
+}
+
+bool TranslatorStorage::loadAndInstallTranslators(const QString &language, const QString &path)
+{
+    auto languageToLoad = language.split("-").at(0);
+
+    bool loadResult = false;
```

Comment:
Function will return result of the last operation, we will need to adjust it here to return false if there has been any error during the process of loading the translations. Alternatively we need to stop loading process after first error. And it will be better to save name of the error file to some kind of variable to get it later by errorMessage method.